src/field/generated_fieldops.cxx

```diff
@@ -0,0 +1,1668 @@
+// This file is autogenerated - see gen_fieldops.py
+#include <bout/mesh.hxx>
+#include <field2d.hxx>
+#include <field3d.hxx>
+#include <globals.hxx>
+#include <interpolation.hxx>
+
+// Do the actual multiplication of Field3D and Field3D
+void autogen_Field3D_Field3D_Field3D_multiplication(BoutReal *__restrict__ result,
+                                                    const BoutReal *__restrict__ lhs,
+                                                    const BoutReal *__restrict__ rhs,
+                                                    int len) {
+
+  for (int i = 0; i < len; ++i) {
+    result[i] = lhs[i] * rhs[i];
+  }
+}
+
+// Provide the C++ wrapper for multiplication of Field3D and Field3D
+Field3D operator*(const Field3D &lhs, const Field3D &rhs) {
```

Comment:
Whilst we haven't typically made use of it, it was possible with the old form to only apply these operators over a specific region (only by hacking the internals though). Probably not a big deal if we lose this ability for now but just to clarify -- is there any way to limit these operations to a region at the moment with this approach?

Reply:
Not currently, but we'll get that back (and hopefully better?) when we move to the SDI-block form